sparse/core.py

```diff
@@ -245,26 +245,43 @@ def __sizeof__(self):
 
     def __getitem__(self, index):
         if not isinstance(index, tuple):
-            index = (index,)
+            if isinstance(index, str):
+                data = self.data[index]
+                idx = np.where(data)
+                coords = []
+
+                for i in range(self.ndim):
+                    coords.append(self.coords[i, idx[0]])
+
+                for i in range(1, np.ndim(data)):
+                    coords.append(idx[i])
+
+                return COO(coords, data.flatten(),
+                           shape=self.shape + self.data.dtype[index].shape,
+                           has_duplicates=self.has_duplicates,
+                           sorted=self.sorted)
+            else:
+                index = (index,)
+
+        last_ellipsis = False
+
         if len(index) - index.count(None) - index.count(Ellipsis) > self.ndim:
             raise IndexError("too many indices for array")
         if index.count(Ellipsis) > 1:
             raise IndexError("an index can only have a single ellipsis ('...')")
-        index = tuple(ind + self.shape[i]  # this fails for newaxis slices
-                      if isinstance(ind, numbers.Integral) and ind < 0
-                      else ind
-                      for i, ind in enumerate(index))
         if any(ind is Ellipsis for ind in index):
             loc = index.index(Ellipsis)
             n = self.ndim - (len(index) - 1 - index.count(None))
-            index = index[:loc] + (slice(None, None),) * n + index[loc + 1:]
-        if all(ind == slice(None) for ind in index):
+            if loc == len(index) - 1:
+                last_ellipsis = True
+            index = index[:loc] + (slice(None),) * n + index[loc + 1:]
+        if len(index) != 0 and all(ind == slice(None) for ind in index):
             return self
-        mask = np.ones(self.nnz, dtype=bool)
+        mask = np.ones(self.nnz, dtype=np.bool)
```

Comment:
This change doesn't do anything

```python
In [5]: np.bool is bool
Out[5]: True

In [6]: np.bool_
Out[6]: numpy.bool_
```

Reply:
My intent was to make it more performant but looks like Numpy treats them the same...

Reply:
Not just Numpy, these are exactly the same objects, the `is` operator is object identity testing in Pyhton.

Reply:
I meant `np.bool_` and `bool`. If you create arrays with both they end up being the same `dtype`.